utils/src/main/java/org/apache/cloudstack/utils/volume/VirtualMachineDiskInfo.java

```diff
@@ -39,4 +38,11 @@ public void setDiskDeviceBusName(String diskDeviceBusName) {
     public void setDiskChain(String[] diskChain) {
         this.diskChain = diskChain;
     }
+
+    public String getControllerFromDeviceBusName() {
+        if (StringUtils.isEmpty(diskDeviceBusName) || !diskDeviceBusName.contains(":")) {
+            return null;
```

Comment:
In either case, we'll need to do a comparison to check if the string is null or empty. Returning an empty string might cover up potential functional issues, while NPEs are bad but can expose the issue without hiding it.